ucm2/Intel/sof-hda-dsp/HiFi-analog-sof.conf

```diff
@@ -0,0 +1,492 @@
+	      SpeakerIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/highpass_100hz_0db_48khz.blob"
+	      SpeakerFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      SpeakerDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/speaker_default.blob"
+	      HeadphoneIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/pass.blob"
+	      HeadphoneFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/passthrough.blob"
+	}
+	False.Define {
+	      SOFIPCVer "ipc3"
+	      PostMixerAnalogPlaybackIIRBytes "EQIIR1.0 eqiir_coef_1"
+	      PostMixerAnalogPlaybackFIRBytes "EQFIR1.0 eqfir_coef_1"
+	      PostMixerAnalogPlaybackDRCBytes "not available"
+	      PostMixerAnalogPlaybackDRCSwitch "not available"
+	      SpeakerIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/highpass_100hz_0db_48khz.blob"
+	      SpeakerFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      SpeakerDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/speaker_default.blob"
+	      HeadphoneIirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_iir/pass.blob"
+	      HeadphoneFirBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/eq_fir/pass.blob"
+	      HeadphoneDrcBlob "${ConfTopDir}/sof/${var:SOFIPCVer}/drc/passthrough.blob"
+	}
+}
+
+# Include blob customizations: sys_vendor / product_name .conf
+# TODO: The 2nd level could be instead product_sku .conf ?
+Define.ProductConfigSOF "/sof/product_configs/${sys:devices/virtual/dmi/id/sys_vendor}/${sys:devices/virtual/dmi/id/product_name}.conf"
+Define.SOFConfFullPath "${ConfTopDir}${var:ProductConfigSOF}"
+
+If.HasProductConfig {
+	Condition {
+		Type Path
+		Mode read
+		Path "${var:SOFConfFullPath}"
+	}
+	True {
+		Include.product-config.File "${var:ProductConfigSOF}"
+	}
+}
+
+If.imicname {
+	Condition {
+		Type ControlExists
+		Control "name='Internal Mic Boost Volume'"
+	}
+	True.Define {
+		FrontMicName "Internal"
+	}
+}
+
+If.fmicname {
+	Condition {
+		Type ControlExists
+		Control "name='Front Mic Playback Switch'"
+	}
+	True.Define {
+		FrontMicName "Front"
+		FrontMicJack "Front Mic Jack"
+	}
+}
+
+If.fmic {
+	Condition {
+		Type String
+		Empty "${var:FrontMicName}"
+	}
+	False {
+		If.mic2 {
+			Condition {
+				Type String
+				String1 "${var:DeviceMic}"
+				String2 "Mic2"
+			}
+			True.Define {
+				DeviceFrontMic "Mic3"
+			}
+			False.Define {
+				DeviceMic "Mic1"
+				DeviceFrontMic "Mic2"
+			}
+		}
+	}
+}
+
+If.lineout {
+	Condition {
+		Type ControlExists
+		Control "name='Line Out Playback Volume'"
+	}
+	True.Define {
+		loctl "Line Out"
+		lovol "Line Out"
+	}
+}
+
+If.linefront {
+	Condition {
+		Type ControlExists
+		Control "name='Front Playback Volume'"
+	}
+	True.Define {
+		LineDevice "1"
+	}
+}
+
+If.hplo {
+	Condition {
+		Type ControlExists
+		Control "name='Headphone+LO Playback Volume'"
+	}
+	True.Define {
+		hpvol "Headphone+LO"
+		lovol "Headphone+LO"
+	}
+}
+
+If.hpjack {
```

Comment:
I don't get how all these variables are set and by what?

Does this require manual setups by the users, or is this information reported by the kernel based on what the codec reports as supported?

Reply:
The If statement syntax was very confusing to me when I started with this work. I think the `If.<condition name>` is just a name for the Condition {}. Maybe it's like a mandatory comment for things done in UCM syntax. It could even be `If.0` as far as I understand. Please correct if I'm wrong @perexg .

Reply:
It's not the syntax that threw me off, rather where those 'linefront' and 'hplo' variables are set and why what?

Reply:
LO = Line Output ; this cryptic name means that Headphone and Line Output shares one volume control.

Reply:
I got the names, just not how you know what the hardware connections are? This is typically buried in nid/verbs.

Reply:
The HDA codecs have NID connection list, thus the routing path is known and those shared volume controls are created when there are missing volume NIDs in this route (to control volumes separately). Some codecs just does not have separate volume controls for all I/O nodes (although, they're rare).